src/wp-includes/link-template.php

```diff
@@ -4688,3 +4688,50 @@ function get_the_privacy_policy_link( $before = '', $after = '' ) {
 
 	return '';
 }
+
+/**
+ * Returns an array of URL hosts which are considered to be internal hosts.
+ *
+ * @since [version]
+ *
+ * @return string[] An array of URL hosts.
+ */
+function wp_internal_hosts() {
+	/**
+	 * Filters the array of URL hosts which are considered internal.
+	 *
+	 * @since [version]
+	 *
+	 * @param array $internal_hosts An array of internal URL hostnames.
+	 */
+	$internal_hosts = apply_filters(
+		'wp_internal_hosts',
+		array(
+			wp_parse_url( home_url(), PHP_URL_HOST ),
+		)
+	);
+	return $internal_hosts;
+}
+
+/**
+ * Determines whether or not the specified URL is of a host included in the internal hosts list.
+ *
+ * @see wp_internal_hosts()
+ *
+ * @since [version]
+ *
+ * @param string $link The URL to test.
+ * @return bool Returns true for internal URLs and false for all other URLs.
+ */
+function wp_is_internal_link( $link ) {
```

Comment:
I would love to see some tests specific to this new function.  

A few specific scenarios:

- Making sure that HTTP and https versions of a URL both work when home_url is https
- Making sure that the filter correctly adds/removes domains